olive/systems/python_environment/pass_runner.py

```diff
@@ -4,41 +4,40 @@
 # --------------------------------------------------------------------------
 import argparse
 import json
+from pathlib import Path
 
+from olive.common.utils import set_tempdir
 from olive.model import ModelConfig
 from olive.passes.olive_pass import FullPassConfig
 
 
 def get_args(raw_args):
     parser = argparse.ArgumentParser(description="Onnx model inference")
 
-    parser.add_argument("--model_json_path", type=str, required=True, help="Path to input model json file")
-    parser.add_argument("--pass_json_path", type=str, required=True, help="Path to pass json file")
-    parser.add_argument("--data_root", type=str, default=None, help="Path to data root")
+    parser.add_argument("--model_config", type=str, required=True, help="Path to input model json file")
+    parser.add_argument("--pass_config", type=str, required=True, help="Path to pass json file")
+    parser.add_argument("--data_root", type=str, required=False, help="Path to data root")
+    parser.add_argument("--tempdir", type=str, required=False, help="Root directory for tempfile directories and files")
     parser.add_argument("--output_model_path", type=str, required=True, help="Path to output model file")
-    parser.add_argument("--output_model_json_path", type=str, required=True, help="Path to output model json file")
+    parser.add_argument("--output_path", type=str, required=True, help="Path to output model json file")
 
     return parser.parse_args(raw_args)
 
 
 def main(raw_args=None):
     args = get_args(raw_args)
 
-    with open(args.model_json_path) as f:
-        model_json = json.load(f)
-    with open(args.pass_json_path) as f:
-        pass_json = json.load(f)
+    set_tempdir(args.tempdir)
 
-    model = ModelConfig.from_json(model_json).create_model()
-    the_pass = FullPassConfig.from_json(pass_json).create_pass()
+    model = ModelConfig.parse_file(args.model_config).create_model()
+    the_pass = FullPassConfig.parse_file(args.pass_config).create_pass()
```

Comment:
I like it.

Reply:
yeah, the `parse_file` method is very useful!